pallets/token-wrapper/src/lib.rs

```diff
@@ -433,12 +433,39 @@ impl<T: Config> TokenWrapperInterface<T::AccountId, T::AssetId, BalanceOf<T>, T:
 			DispatchError,
 		>>::get_by_id(into_pool_share_id)?;
 
-		T::Currency::transfer(
-			from_currency_id,
-			&from,
-			&Self::get_fee_recipient(&asset_details.name),
-			Self::get_wrapping_fee(amount, into_pool_share_id)?,
-		)?;
+		let fee_recipient_account = Self::get_fee_recipient(&asset_details.name);
+		let wrapping_fee = Self::get_wrapping_fee(amount, into_pool_share_id)?;
+
+		// check if the amount is lower than ED
+		if wrapping_fee < T::Currency::minimum_balance(from_currency_id) {
+			// in this case transfer only if the recipient account already has ED
+			let current_balance_fee_recipient =
+				T::Currency::total_balance(from_currency_id, &fee_recipient_account);
```

Comment:
I do believe even if the `fee_recipient_account`'s balance is greater than than the `minimum_balance` and the `wrapping_fee` is less than `minimum_balance` (the `ED` value) it will fail.